src/Model/Item.php

```diff
@@ -1403,7 +1408,7 @@ public static function storeForUserByUriId(int $uri_id, int $uid, array $fields
 		$is_reshare = ($item['gravity'] == GRAVITY_ACTIVITY) && ($item['verb'] == Activity::ANNOUNCE);
 
 		if ((($item['gravity'] == GRAVITY_PARENT) || $is_reshare) &&
-			DI::pConfig()->get($uid, 'system', 'accept_only_sharer') &&
+			DI::pConfig()->get($uid, 'system', 'accept_only_sharer') === self::COMPLETION_NONE &&
```

Comment:
The pconfig value will most likely contain a boolean value, so comparing it via `===` wouldn't work, wouldn't it?

Reply:
The pconfig value was stored as an integer even when it was a boolean flag, so there's no issue there.

Reply:
Ping @annando

Reply:
Sorry, I was busy with the other task.